cmake/range-v3.cmake

```diff
@@ -1,38 +1,6 @@
-include(ExternalProject)
-
-if (${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten")
-    set(RANGE_V3_CMAKE_COMMAND emcmake cmake)
-else()
-    set(RANGE_V3_CMAKE_COMMAND ${CMAKE_COMMAND})
-endif()
-
-set(prefix "${PROJECT_BINARY_DIR}/deps")
-set(RANGE_V3_INCLUDE_DIR "${prefix}/include")
-
-ExternalProject_Add(range-v3-project
-    PREFIX "${prefix}"
-    DOWNLOAD_DIR "${PROJECT_SOURCE_DIR}/deps/downloads"
-    DOWNLOAD_NAME range-v3-0.12.0.tar.gz
-    URL https://github.com/ericniebler/range-v3/archive/0.12.0.tar.gz
-    URL_HASH SHA256=015adb2300a98edfceaf0725beec3337f542af4915cec4d0b89fa0886f4ba9cb
-    CMAKE_COMMAND ${RANGE_V3_CMAKE_COMMAND}
-    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
-               -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-               -DBUILD_TESTING=OFF
-               -DRANGES_CXX_STD=${CMAKE_CXX_STANDARD}
-               -DRANGE_V3_DOCS=OFF
-               -DRANGE_V3_EXAMPLES=OFF
-               -DRANGE_V3_TESTS=OFF
-               -DRANGES_BUILD_CALENDAR_EXAMPLE=OFF
```

Comment:
I think we still want to keep some of these off. For example [`RANGES_BUILD_CALENDAR_EXAMPLE` defaults to `ON`](https://github.com/ericniebler/range-v3/blob/0.12.0/cmake/ranges_options.cmake#L10), but sounds like pure compilation overhead for us.

`RANGE_V3_DOCS`, `RANGE_V3_EXAMPLES`, `RANGE_V3_TESTS` as well, though that depends on whether this is now treated as a "subproject" or not.

Reply:
We are not building the library anymore, we're just using the header only implementation now.